components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -446,7 +447,62 @@ public API getAPI(APIIdentifier identifier) throws APIManagementException {
 
             API api = APIUtil.getAPIForPublishing(apiArtifact, registry);
             APIUtil.updateAPIProductDependencies(api, registry);
+            api.setSwaggerDefinition(getOpenAPIDefinition(identifier));
```

Comment:
Is there any place where this getAPI() method getting called? Ideally this method should not get executed.

Reply:
no, this change is coming from old code, no uses for Revisions now